packages/beacon-node/src/metrics/metrics/libp2p.ts

```diff
@@ -0,0 +1,84 @@
+import {Libp2p} from "libp2p";
+import {GaugeExtra} from "../utils/gauge.js";
+import {RegistryMetricCreator} from "../utils/registryMetricCreator.js";
+
+export type ILibp2pMetrics = ReturnType<typeof createLibp2pMetrics>;
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function createLibp2pMetrics(libp2p: Libp2p, register: RegistryMetricCreator) {
+  const libp2pMetrics = libp2p.metrics;
+  if (libp2pMetrics === undefined) {
+    return;
+  }
+
+  const metrics: Record<string, GaugeExtra<string>> = {};
+
+  const ensureLibp2pMetrics = async (): Promise<void> => {
+    // protocol metrics
+    const protocols = libp2pMetrics.getProtocols();
+    protocols.forEach((protocol) => {
+      const protocolStat = libp2pMetrics.forProtocol(protocol);
+      if (protocolStat === undefined) {
+        return;
+      }
+
+      // create metric if undefined
+      if (metrics[protocol] === undefined) {
```

Comment:
shouldn't we just add `protocol` in `labelNames` and simplify this

Reply:
I think not, bc they split out the  direction. eg: would be helpful to see if we're spammed with identify requests, but not responding.

Reply:
Oh, just add `protocol` as an additional label. Yes good idea. Lets tackle metrics cleanup in a separate PR.